stdlib/messaging/activemq-artemis/src/main/ballerina/artemis/listener.bal

```diff
@@ -19,20 +19,26 @@
 public type Listener object {
     *AbstractListener;
     private Session session;
-    private boolean anonymousSession;
 
-    public function __init(Session | URLConfiguration sesssionOrURLConfig) {
-        if (sesssionOrURLConfig is Session) {
-            self.session = sesssionOrURLConfig;
+    public function __init(Session | EndpointConfiguration sessionOrEndpointConfig) {
+        if (sessionOrEndpointConfig is Session) {
+            self.session = sessionOrEndpointConfig;
         } else {
-            Connection connection = new("tcp://" + sesssionOrURLConfig.host + ":" + sesssionOrURLConfig.port);
-            self.session = new(connection, config = { username: sesssionOrURLConfig["username"],
-                    password: sesssionOrURLConfig["password"] });
-            self.anonymousSession = true;
+            Connection connection;
+            var secureSocket = sessionOrEndpointConfig["secureSocket"];
+            if (secureSocket is SecureSocket) {
+                connection = new(parseUrl(sessionOrEndpointConfig), config = {secureSocket: secureSocket});
+            } else {
+                connection = new(parseUrl(sessionOrEndpointConfig));
+            }
+            self.session = new(connection, config = { username: sessionOrEndpointConfig["username"],
+                    password: sessionOrEndpointConfig["password"],
+                    autoCommitSends: false, autoCommitAcks:  false });
+            self.session.anonymousSession = true;
         }
     }
     public function __start() returns error? {
-        return self.start();
+        return self. start();
```

Comment:
Shall we remove additional space here

Reply:
This is an issue with VSCode formatting. Created an [issue](https://github.com/ballerina-platform/ballerina-lang/issues/15264) and I have fixed it manually here.